### PR TITLE
Adds a framework for airlocks changing access depending on security level.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
This allows you to have a door that opens with no access req at code green but locks to another access on other security levels.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -2059,6 +2064,8 @@ About the new airlock wires panel:
 		to_chat(user, bracer.health)
 	if(p_open)
 		to_chat(user, "\The [src]'s maintenance panel has been unscrewed and is hanging open.")
+	if(islist(access_override_by_level))
+		to_chat(user, SPAN_NOTICE("This airlock changes access requirement depending on the level."))
```
Comment (FluffyGhoster) [inline code/game/machinery/doors/airlock.dm:2068]:
```suggestion
		to_chat(user, SPAN_NOTICE("This airlock changes access requirements depending on the alert level."))
```

File: code/modules/effects/map_effects/door_helper.dm [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+
+/obj/effect/map_effect/door_helper/unres
+	icon_state = "unres_door"
+
+/obj/effect/map_effect/door_helper/unres/modify_door(obj/machinery/door/D)
+	D.unres_dir ^= dir
+
+/obj/effect/map_effect/door_helper/level_access
+	icon_state = "level_door"
+	/// Example of an appropriate way to set this: list("red" = list(1, 2))
```
Comment (FluffyGhoster) [inline code/modules/effects/map_effects/door_helper.dm:23]:
```suggestion

	/// Example of an appropriate way to set this: list("red" = list(1, 2))
```
Should also say what the var does, before showing examples on how to set it

**Review by Arrow768 — CHANGES_REQUESTED** [review]
This needs to work with req_one_access as well.

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**